### PR TITLE
Fix #5088:SCIM2 user listing with an offset for a secondary user store is not working properly

### PR DESCRIPTION
**Pull request by IsurangaPerera**
- Fix wso2/product-is#5088
The issue was caused by non-scim users user being added to the result set. Since they don't have a scim id it returns "invalidValue","detail":"Required attribute id is missing in the SCIM Object." response.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -538,7 +539,7 @@ private void retriveSCIMUsers(List<Object> users, String[] userNames, List<Strin
                             + "Including user : " + userName + " in the response.");
                 }
                 User scimUser = this.getSCIMUser(userName, requiredClaims, scimToLocalClaimsMap);
-                if (scimUser != null) {
+                if (scimUser != null && StringUtils.isNotBlank(scimUser.getId())) {
```
Comment (malithie) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Shouldn't this be handled within the getSCIMUser it self similar to [1].
That's seems to be the core method being used to build up the user from SCIM schema and being referred in other places. Fixing only this place will not fix any other places where the same issue could occur

[1] https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/bb3cbb2d11fa4e2940c3fec29f662a0c21f1c21f/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java#L2157

Comment (IsurangaPerera) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Fixed

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -2055,6 +2056,10 @@ private User getSCIMUser(String userName, List<String> claimURIList, Map<String,
             Map<String, String> attributes = SCIMCommonUtils.convertLocalToSCIMDialect(userClaimValues,
                     scimToLocalClaimsMap);
 
+            if (!attributes.containsKey(SCIMConstants.CommonSchemaConstants.ID_URI)) {
+                return scimUser;
+            }
```
Comment (malithie) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: With this fix there's a possibility to return null from this method.
Make sure invocations of this method handles that

Comment (IsurangaPerera) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: it's already handled

**Pull request merged**